project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: default
+title: 2023 Q2 Hyperledger Sawtooth
+parent: 2023
+grand_parent: Project Updates
+---
+
+# Project Health
+
+Sawtooth is going through a transition from Transact being separated and the idea that Sawtooth 2.0 would use https://github.com/Cargill/splinter . The last two monthly calls have looked at merging Transact and some of the libraries Bitwise and Blockchain TP created with the current 1.3.x branch and renaming it "main".  That would allow the 1.x branch to continue as several companies are using it exclusively for their clients(including Taekion & BlockchainTP)as well as not be dependent on a non-Hyperledger project - Splinter. The advantage is having a blockchain that can easily using different consensus mechanisms easily for different workload situations. 
```

Comment:
While we are not going forward with the more drastic redesign that was intended for 2.x, there was nevertheless some good rust work that was done which is not dependent on the full architecture.  Notably, some of this code made its way into Transact and is relatively straightforward to port (in theory).

I cannot comment on releases of Transact or Splinter projects, other than to remind that they each were originally closely related to sawtooth.

Reply:
I made the suggestion to possibly rename the '1-3' branch to 'main' based on the idea that if 1.3.x became the focus of new development (based on discussions), it makes sense to have a unified direction. None of those contributions has actually occurred, so there is no current reason to rename the branch or think it will be more active than the current main branch.

Transact is in maintenance mode (project status has changed), but has functionally been merged into Sawtooth lib.

Splilnter's roadmap is here: https://splinter.dev/community/planning/roadmap.html, though development is slow.

Reply:
I'm also having a lot of trouble understanding this paragraph. Based on the clarifications above and elsewhere, I suggest the paragraph needs a re-write to describe the different threads:
- Transact has been successfully merged into Sawtooth, due to the project going into the Dormant mode. But there are still contributions to the Transact code base that need to be ported into Sawtooth continuously
- The 1.x branches are the current focus of development activities, and expected to be for the foreseeable future.
- Sawtooth 2.0 is a redesign and will be based on Splinter, which is a component from outside of Hyperledger, but was created with Sawtooth support in mind. At the moment the community's interest for the 2.0 development is low.
- A proposal was made to consider making the 1.3.x branch the new main, to replace the 2.0 work that the main was intended for, to reflect the focus of the community

I hope I was able to capture the various points throughout the discussions, but I'm sure I may have misinterpreted something entirely